### PR TITLE
Fix undefined serialization in URIs

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

Fixes in-app QR code generation for user sharing such that undefined URI data values (like currency code) are not included in the generated URI. 

### Other changes

_Describe any minor or "drive-by" changes here._

### Tested

Added unit tests.

### Related issues

- Fixes scanning issues seen in onboarding

### Backwards compatibility

URIs are now always URL encoded for interoperability.

### CODE REVIEW

File: packages/mobile/src/qrcode/schema.ts [hunk 1]
```diff
@@ -29,13 +29,15 @@ export const uriDataFromJson = (obj: object): UriData => {
...
 
 enum UriMethod {
   pay = 'pay',
 }
 
 export const urlFromUriData = (data: Partial<UriData>, method: UriMethod = UriMethod.pay) => {
-  const params = new URLSearchParams(Object(data))
-  return `celo://wallet/${method.toString()}?${params.toString()}`
+  const params = new URLSearchParams(
+    Object(Object.entries(data).filter(([k, v], _) => v !== undefined))
```
Comment (annakaz) [inline packages/mobile/src/qrcode/schema.ts]: Nit: Can you add a comment I find this difficult to read

**Review by annakaz — APPROVED** [review]
Approving to get this in the latest build / a fix out as soon as we can!

Have you tested on device? If not, I can test locally using two devices

**Comment by annakaz** [comment]
Tested QR send on two devices 💯

File: packages/mobile/src/qrcode/schema.ts (hunk 1 above)
Comment (jmrossy) [reply packages/mobile/src/qrcode/schema.ts]: +1, or consider breaking this apart.

**Pull request merged**